File: client/android/UseCase/Auth/Session/ResetRecoveryUseCase.kt
package app.mobiling.client.usecase.auth.session

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.contract.auth.session.ResetRecoveryRequest
import app.mobiling.client.data.auth.session.AuthSessionGateway

class ResetRecoveryUseCase(
    private val gateway: AuthSessionGateway,
) {
    suspend operator fun invoke(request: ResetRecoveryRequest): AuthSessionPayload =
        gateway.resetRecovery(request)
}

File: client/ios/UseCase/Auth/Session/ResetRecoveryUseCase.swift
public struct ResetRecoveryUseCase {
    private let gateway: AuthSessionGateway

    public init(gateway: AuthSessionGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(request: ResetRecoveryRequest) async throws -> AuthSessionPayload {
        try await gateway.resetRecovery(request: request)
    }
}
